src/Momento.Sdk/Internal/Utils.cs

```diff
@@ -125,4 +125,33 @@ public int GetHashCode(T obj)
     /// so comparisons operate on byte-array content instead of reference.
     /// </summary>
     public static StructuralEqualityComparer<byte[]> ByteArrayComparer = new();
+
+    /// <summary>
+    /// Thrown an exception if the end index is larger than the start index.
+    /// </summary>
+    /// <param name="startIndex">Starting inclusive index of operation.</param>
+    /// <param name="endIndex">Ending exclusive index of operation.</param>
+    /// <exception cref="Momento.Sdk.Exceptions.InvalidArgumentException"><paramref name="endIndex"/> is less than or equal to <paramref name="startIndex"/></exception>
+    public static void ValidateStartEndIndex(int? startIndex, int? endIndex)
+    {
+        if (startIndex.HasValue && endIndex.HasValue)
+        {
+            if (startIndex >= 0 && endIndex >= 0 && startIndex >= endIndex)
+            {
+                throw new Momento.Sdk.Exceptions.InvalidArgumentException("endIndex (exclusive) must be larger than startIndex (inclusive)");
+            }
+            if (startIndex < 0 && endIndex < 0 && startIndex >= endIndex)
+            {
+                throw new Momento.Sdk.Exceptions.InvalidArgumentException("endIndex (exclusive) must be larger than startIndex (inclusive)");
+            }
+            else
```

Comment:
supernit: I don't think that either this else clause or the one below it are technically necessary